### PR TITLE
🔥 Escape all HTML inserted into title tags

### PR DESCRIPTION
**Pull request by DerekNonGeneric**
Untested, but hopefully the CI will be able to do that.

Fixes: #3472
Refs:
> Escaping works by piping the variable through the |e filter:
>```jinja
> {{ user.username|e }}
> ```
> — [Template Designer Documentation &#8212; Jinja Documentation (2.11.x)](https://jinja.palletsprojects.com/en/2.11.x/templates/#working-with-manual-escaping)

### CODE REVIEW

**Comment by matthiasrohmer** [comment]
Ha, I was about to comment that this wouldn't fix it 😄 Your second commit should do it... but there's a problem: https://amp.dev/documentation/components/amp-autocomplete/?format=websites (or any other component like `amp-subscription` Malte was trying to mention) would then have `&lt;amp-accordion&gt;` in their title tag which wouldn't be unescaped by browsers. At least that's what my quick check showed by setting `document.title` in the browser console.

File: examples/source/1.components/amp-sticky-ad.html [hunk 1]
```diff
@@ -3,13 +3,13 @@
 author: kul3r4
 
 --->
-  <!--
+<!--
```
Comment (matthiasrohmer) [inline examples/source/1.components/amp-sticky-ad.html]: Please revert the changes to this file as it's unrelated.

File: frontend/templates/layouts/example-pages/amp-ads-page.j2 [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <head>
   <meta charset="utf-8">
 {% if not example.has_tag_in_head('title') %}
-  <title>{{export.title|e}} - {{_('Example')}} {{example.index}}</title>
+  <title>{{export.title|e}} - {{_('Example')|e}} {{example.index|e}}</title>
```
Comment (matthiasrohmer) [inline frontend/templates/layouts/example-pages/amp-ads-page.j2]:
```suggestion
  <title>{{export.title|e}} - {{_('Example')}} {{example.index}}</title>
```

It's safe to assume that the result of `_('Example')` will not contain any HMTL chars. The same for `example.index`.

**Comment by DerekNonGeneric** [comment]
Haha, you almost caught me. :) I propose we remove these symbols as you can see in my fixup commit. Thoughts?

**Comment by DerekNonGeneric** [comment]
I was not able to replicate your other point about the browser showing unescaped title.


On my local machine with the change (in the Chrome console REPL):

```javascript
document.title
"Documentation: <amp-subscriptions> - amp.dev"
```

File: frontend/templates/views/partials/structured-data.j2 [hunk 3]
```diff
@@ -60,7 +60,7 @@
 {% set title_prefix = _('Success Story: ') %}
 {% endif %}
 
-<title>{{ title_prefix }}{{ title }} - amp.dev</title>
+<title>{{title_prefix|e}}{{title|e}} - amp.dev</title>
```
Comment (matthiasrohmer) [inline frontend/templates/views/partials/structured-data.j2]:
```suggestion
<title>{{title_prefix}}{{title}} - amp.dev</title>
```

What's inside the `<title>` tag shouldn't be escaped. As I mentioned I think they wouldn't be unescaped when shown in the browser title bar but are gracefully handled by Google for example. That's why we initially used `<` over `&lt;`.

File: pages/content/amp-dev/documentation/guides-and-tutorials/learn/spec/release-schedule.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ A new release of AMP is pushed to all AMP pages every week on Tuesday. **Once a
 
 The AMP runtime and extensions are provided through a variety of different _release channels_. Each channel serves a purpose for developers and for the AMP HTML Project itself. See the [release cadence section](#release-cadence) for a more detailed breakdown of how and when code from the [`ampproject/amphtml`](https://github.com/ampproject/amphtml) repository makes it into release builds.
 
-To determine if a PR has been included in any of the following release channels, look for the GitHub labels _PR Use: In Canary_, _PR Use: In Production_, or _PR Use: In LTS_ (see the section on [determining if your change is in a release](#determining-if-your-change-is-in-a-release) for more details).
+To determine if a PR has been included in any of the following release channels, look for the GitHub labels _PR Use: In Canary_, _PR Use: In Production_, or _PR Use: In LTS_ (see the section on [determining if your change is in a release](#Determining-if-your-change-is-in-a-release) for more details).
```
Comment (matthiasrohmer) [inline pages/content/amp-dev/documentation/guides-and-tutorials/learn/spec/release-schedule.md]: Please reset as unrelated.

File: pages/shared/data/blog.yaml [hunk 5]
```diff
@@ -1,5 +1,11 @@
 entries:
 
+- title: "Blog"
```
Comment (matthiasrohmer) [inline pages/shared/data/blog.yaml]: Please reset as unrelated.

**Comment by matthiasrohmer** [comment]
> I was not able to replicate your other point about the browser showing unescaped title.
> 
> On my local machine with the change (in the Chrome console REPL):
> 
> ```js
> document.title
> "Documentation: <amp-subscriptions> - amp.dev"
> ```

Ups. Was reviewing while you've been working on it. I only did a quick check by executing `document.title = "&lt;";` in the REPL. That didn't print as `<`.  But could als be that this isn't the most valid test.

**Comment by DerekNonGeneric** [comment]
> But could also be that this isn't the most valid test.

The HTML entities get converted once the source is parsed + evaluated, so once it's in the DOM node it will have already been converted. Unless the source text is being scraped without ever being evaluated, this shouldn't be an issue. Pretty sure the Google crawler uses headless Chrome for all of this. If you really want me to remove escaping the title tags, just say the word and I'll remove those filters.
